OneSignalSDK/onesignal/in-app-messages/src/main/java/com/onesignal/inAppMessages/internal/InAppMessagesManager.kt

```diff
@@ -112,6 +112,11 @@ internal class InAppMessagesManager(
             Logging.debug("InAppMessagesManager.setPaused(value: $value)")
             _state.paused = value
 
+            // If paused is true and an In-App Message is showing, dismiss it
+            if (value && _state.inAppMessageIdShowing != null) {
```

Comment:
Does this always run on the main thread? I noticed that the block below does suspendifyOnThread

Reply:
Updates:
- Confirmed that pausing from the click handler also successfully dismisses currently-display IAM, remaining lifecycle listeners fire, and following queued IAMs do not display
- Added a coroutine to ensure that dismissal always fires on main thread